governator-core/src/main/java/com/netflix/governator/internal/PreDestroyMonitor.java

```diff
@@ -172,60 +209,80 @@ public void addScopeBindings(Map<Class<? extends Annotation>, Scope> bindings) {
      */
     @Override
     public void close() throws Exception {
-        if (scopeCleaner.close()) { // executor thread to exit processing loop
+        if (scopeCleaner.close()) { // executor thread to exit processing loop            
             LOGGER.info("closing PreDestroyMonitor...");
-
-            for (Callable<Void> action : cleanupActions) {
-                action.call();
+            synchronized(cleanupActions) {
+                List<Map.Entry<Object, UnscopedCleanupAction>> actions = new ArrayList<>(cleanupActions.entrySet());
```

Comment:
UnscopedCleanupAction already implements Comparable.  We can simplify this code using streams,
```
actions.entrySet()
		.stream()
		.sorted(Map.Entry.comparingByValue())
		.forEach(
				action -> action.getValue().call(action.getKey())
		);
```

Reply:
hadn't seen Map.Entry.comparingByValue()

Reply:
done